inet/vibe/inet/url.d

```diff
@@ -19,11 +19,75 @@ import std.exception;
 import std.string;
 import std.traits : isInstanceOf;
 import std.ascii : isAlpha, isASCII, toLower;
-import std.uri: encode;
+import std.uri: decode, encode;
 
 import core.checkedint : addu;
 
 
+/** Parses a user-entered URL with relaxed parsing rules.
+
+	Unlike `URL.parse`, this allows the URL to use special charaters as part of
+	the host name and path, automatically emplying punycode or percent-encoding
+	to convert this to a valid URL.
+
+	Also, relative URL references are supported by supplying a base URL.
+
+	Params:
+		url = String representation of the URL
+		base_url = Base
+*/
+URL parseUserURL(string url, URL base_url = URL.init)
+{
+	// test if this is a URL or a relative (path) reference
+	auto sidx = url.indexOf(':');
+	if (sidx < 0) {
```

Comment:
I would say that the fact that it's an URL is the special case, so use an early return:
```D
if (sidx >= 0)
    return URL(url, false);
```

This way we can avoid putting 90% f the function in an `if`.
Also, why do we pass `false` as `decode` ? It differs from what the ctor does by default and thus violates the principle of least surprise.

Reply:
I just did an incremental change here to keep the diff small. Ultimately, I'd expect to have the whole encoding logic in this function instead of in a special constructor of `URL`.